addons/godot_rl_agents/sensors/sensors_3d/GridSensor3D.gd

```diff
@@ -38,29 +38,55 @@ class_name GridSensor3D
 		cell_height = value
 		_update()	
 
-@export_range(1, 21, 2, "or_greater") var grid_size := 3:
-	get: return grid_size
+@export_range(1, 21, 2, "or_greater") var grid_size_x := 3:
+	get: return grid_size_x
 	set(value):
-		grid_size = value
+		grid_size_x = value
 		_update()
 
-var _obs_buffer : PackedByteArray
-var _box_shape : BoxShape3D
-var _collision_mapping : Dictionary
-var _n_layers_per_cell : int
+@export_range(1, 21, 2, "or_greater") var grid_size_z := 3:
+	get: return grid_size_z
+	set(value):
+		grid_size_z = value
+		_update()
+
+var _obs_buffer: PackedFloat64Array
+var _box_shape: BoxShape3D
+var _collision_mapping: Dictionary
+var _n_layers_per_cell: int
 
+var _highlighted_box_material: StandardMaterial3D
+var _standard_box_material: StandardMaterial3D
 
 func get_observation():
 	return _obs_buffer
-
-
+	
 func _update():
 	if Engine.is_editor_hint():
-		_spawn_nodes()	
-
+		if is_node_ready():
+			_spawn_nodes()	
 
 func _ready() -> void:
-	_spawn_nodes()
+	if Engine.is_editor_hint():	
+		assert(get_child_count() > 0,
```

Comment:
This is for covering a special case, if e.g. nodes are manually removed from the grid sensor before saving the scene, then reopening it in editor. May not be necessary.